### PR TITLE
Implement components v2

### PR DESCRIPTION
**Pull request by khengschew**
* Create components and subcomponents for Summary, Good To Know, and Reviews sections
* Style components using styled-components
* Add unit tests for components
* Update generation scripts to use s3 resources

### CODE REVIEW

File: client/src/components/gtk/gTKArticleList.jsx [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+    }
+
+    return articleList;
+  }
+}
+
+GTKArticleList.propTypes = {
+  name: PropTypes.string.isRequired,
+  id: PropTypes.number.isRequired,
+  articles: PropTypes.arrayOf(PropTypes.shape({
```
Comment (qtrangdo) [inline client/src/components/gtk/gTKArticleList.jsx:96]: proptypes incomplete. Should include the "isRequired" for the keys within articles shape too

File: client/src/components/gtk/gTKOrderList.jsx [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+
+const GTKOrderList = (props) => {
+  const { whatToOrder } = props;
+
+  let orderList = '';
+  if (whatToOrder && whatToOrder.length) {
+    orderList = (
+      <div id="GTKOrderList">
+        <GTKHeader>What To Order</GTKHeader>
+        {whatToOrder
```
Comment (qtrangdo) [inline client/src/components/gtk/gTKOrderList.jsx:14]: unnecessary line break and parens

File: client/src/components/reviews/reviewItem.jsx [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
+
+export const SeeMore = styled.span`
+  color: #9b9b9b;
+`;
+
+class ReviewItem extends React.Component {
+  constructor(props) {
+    super(props);
+
+    const { review } = props;
```
Comment (qtrangdo) [inline client/src/components/reviews/reviewItem.jsx:61]: this line serve no purpose

File: client/src/components/reviews/reviewItem.jsx [hunk 4]
```diff
@@ -0,0 +1,135 @@
...
+            {dispDescription === review.description ? '' : <SeeMore>... See More</SeeMore>}
+          </div>
+        </ReviewContent>
+      </ReviewItemDiv>
+    );
+
+    return reviewDiv;
+  }
+}
+
```
Comment (qtrangdo) [inline client/src/components/reviews/reviewItem.jsx:123]: make sure to complete PropTypes

File: client/src/components/summary/summary.jsx [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+        this.setState({
+          restaurant: data,
+        });
+      });
+  }
+
+  render() {
+    const { restaurant } = this.state;
+
+    return (
```
Comment (qtrangdo) [inline client/src/components/summary/summary.jsx:52]: No handling for when restaurant is {} ?
wouldn't restaurant.(properties) be undefined at this point?

**Review by qtrangdo — COMMENTED** [review]
_ I like the way you handle conditional statement in rendering a component
- static functions are new to me and I thought that's very clever to contain materials in 1 function
- Some incomplete proptypes there but nothing major.
- Handling empty state data for summary component?
- Again, really clean code.

File: client/src/components/gtk/gTKArticleList.jsx (hunk 1 above)
Comment (khengschew) [reply client/src/components/gtk/gTKArticleList.jsx:96]: Solution: refactor to not need propTypes (by handling the "no data" case with empty string like in previous components).

File: client/src/components/gtk/gTKOrderList.jsx (hunk 2 above)
Comment (khengschew) [reply client/src/components/gtk/gTKOrderList.jsx:14]: Eslint pickiness

File: client/src/components/reviews/reviewItem.jsx (hunk 4 above)
Comment (khengschew) [reply client/src/components/reviews/reviewItem.jsx:123]: Solution: refactor to not need propTypes (by handling the "no data" case with empty string like in previous components).

**Pull request merged**